src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -196,26 +196,14 @@ Vector MotionController::determineLinearVelocityFromVelocity(
     const double max_acceleration_meters_per_second_squared, const double delta_time,
     const double max_speed_meters_per_second)
 {
-    double velocity_to_add = max_acceleration_meters_per_second_squared * delta_time;
```

Comment:
My concern here is that all time dependence is removed from the motion controller. It also looks like new_velocity_x/y changed from an accumulative robot speed to a delta. Would you be able to explain why you made these changes?

Reply:
The velocity planner was slow, to a point where when you give a really high velocity, when the robot was initially at zero, it would never reach even 0.5 m/s let alone the max speed, it would increment with very small amounts until it reached +0 increments. So the bot would just wobble back and forth and stop and never actually reach the final position. 

I don't know if I was using the velocity planner incorrectly, but that was what I observed, removing time dependency and replacing it with the delta velocity meant that if the robot was at 0 it would quickly get up to speed, and will adjust based on the difference between its speed and the requested speed. 

I didn't take any controls course so I am not sure if this change makes sense, but I was referring to this link, slide 11: https://courses.cs.washington.edu/courses/cse466/11au/calendar/10-control-posted.pdf

Reply:
Hmm, the only concern here is that we're not explicitly making sure that the robot doesn't accelerate faster then is physically possible. If this looks reasonable in sim though, it's definitely a solid improvement, so I'd perfectly happy with this going in (there will likely be a lot of simulator re-work next year anyhow).

Reply:
Hmm, the only possible concern here is that we're not explicitly making sure that the robot doesn't accelerate faster then is physically possible. If this looks reasonable in sim though, it's definitely a solid improvement, so I'd perfectly happy with this going in (there will likely be a lot of simulator re-work next year anyhow).